### DISCUSSION TITLE
Layer of Threejs on top of map

### DISCUSSION THREAD

**Discussion by hugosoli** [heremaps/harp.gl#2325] [Q&A] [answered]
Hi everybody. Onre more questions. I am building an art work using data of Mexico City. I need to have the map and at the same time have a bunch of text on the screen. The traditional way to have it is by "div" html on top of the harpgl. At least the example 
https://www.harp.gl/docs/master/examples/#object-picking.html shows that aproach.
What if I want to have threejs text (and other objects) that are unmutable to movement. The threejs example https://threejs.org/examples/#webgl_sprites has to scenes and two cameras. One perspective camera for the scene, and one orrographic scene-camera for other materials. I would love to have the same situation. I tried for many many hours but I faild. Any hint to have the harpgl ans also another ortograaphic scene on top would be appreciate it.
Warm regards
Hugo

**Comment by nzjony** [comment, marked as answer]
@hugosoli , hmm, it should be possible. 

Check out: https://www.harp.gl/docs/master/examples/#rendering_post-effects_all.html and also: https://github.com/heremaps/harp.gl/blob/master/%40here/harp-mapview/lib/composing/UnrealBloomPass.ts it uses an Orthographic camera as well.. does that help?

Edit: best answer: https://github.com/heremaps/harp.gl/commit/baec6991cdb947e0b84c73a1652fcdb902d84ef4

**Reply by hugosoli** [reply]
Hi!
I studied such codes and try to extract a minimal example. I Fail. I am stuck. I was assuming something like this (in general terms) could works but is not. My hypothesis is that this DO happen but it is overwriten in the regular "render" which I can not overwrite.
The rendering_post-effect has many prebuild effects but I can not figure out how to add a new effect into the chan.
The composing/UnrealBloomPass.ts seems a way to go but I cant understand how to extend it.

my idea::

  const cameraOrtho = new THREE.OrthographicCamera(- width / 2, width / 2, height / 2, - height / 2, 1, 10);
  cameraOrtho.position.z = 5;
  const sceneOrtho = new THREE.Scene();
  mapView.renderer.clear();
  mapView.renderer.render(mapView.scene, mapView.camera);
  mapView.renderer.clearDepth();
  mapView.renderer.render(sceneOrtho, cameraOrtho);

Hope anybody with more experience in the harpgl stack could give advice. Cheers

**Reply by nzjony** [reply]
@hugosoli , see: https://github.com/heremaps/harp.gl/commit/baec6991cdb947e0b84c73a1652fcdb902d84ef4

It should look like this:
<img width="1680" alt="Screenshot 2021-10-19 at 15 56 19" src="https://user-images.githubusercontent.com/2533428/137924904-45f4c430-8973-4d1f-9081-75dac5d3a03e.png">

**Comment by hugosoli** [comment]
Nice! Really happy this is exactly what I needed. Tons of thanks.

**Reply by nzjony** [reply]
de nada!
